brainiak/fcma/classifier.py

```diff
@@ -459,16 +457,12 @@ def fit(self, X, y, num_training_samples=None):
             self.
         """
         time1 = time.time()
-        if not isinstance(X, tuple):
-            X = (X, X)
-        assert len(X) == 2, \
-            'there must be two parts for correlation computation'
-        X1 = X[0]
-        X2 = X[1]
-        assert len(X1) == len(y), \
-            'the number of samples does not match the number labels'
-        assert len(X1) == len(X2), \
-            'the numbers of samples in X1 and X2 are not identical'
+        assert len(X) == len(y), \
+            'the number of samples must be equal to the number of labels'
+        for x in X:
+            assert len(x) == 2, \
+                'there must be two parts for each correlation computation'
+        X1, X2 = zip(*X)
```

Comment:
Note that `X1` and `X2` will be iterators. In case you want them to be actual lists later, you must create lists based on them.